lib/RTCSession.js

```diff
@@ -626,45 +626,53 @@ module.exports = class RTCSession extends EventEmitter
     // TODO: This may throw an error, should react.
     this._createRTCConnection(pcConfig, rtcConstraints);
 
-    // If a local MediaStream is given use it.
-    if (mediaStream)
-    {
-      userMediaSucceeded.call(this, mediaStream);
-    }
-    // If at least audio or video is requested prompt getUserMedia.
-    else if (mediaConstraints.audio || mediaConstraints.video)
-    {
-      this._localMediaStreamLocallyGenerated = true;
-      navigator.mediaDevices.getUserMedia(mediaConstraints)
-        .then(userMediaSucceeded.bind(this))
-        .catch((error) =>
+    Promise.resolve()
+      // Handle local MediaStream.
+      .then(() =>
+      {
+        // A local MediaStream is given, use it.
+        if (mediaStream)
         {
-          userMediaFailed.call(this, error);
-
-          debugerror('emit "getusermediafailed" [error:%o]', error);
+          return mediaStream;
+        }
 
-          this.emit('getusermediafailed', error);
-        });
-    // Otherwise don't prompt getUserMedia.
-    }
-    else
-    {
-      userMediaSucceeded.call(this, null);
-    }
+        // Audio and/or video requested, prompt getUserMedia.
+        else if (mediaConstraints.audio || mediaConstraints.video)
+        {
+          this._localMediaStreamLocallyGenerated = true;
 
-    // User media succeeded.
-    function userMediaSucceeded(stream)
-    {
-      if (this._status === C.STATUS_TERMINATED) { return; }
+          return navigator.mediaDevices.getUserMedia(mediaConstraints)
+            .catch(() =>
+            {
+              request.reply(480);
```

Comment:
You are not calling `debugerror('emit "getusermediafailed" [error:%o]', error);` anymore (which is needed by the callstats.io JsSIP plugin.

Reply:
Good catch.